### ISSUE TITLE
I have reorganized the code and tested it recently. (08/27/2023)

### ISSUE THREAD

**Issue by Bayi-Hu** [git-disl/BERT4ETH#5]
The code should be able to reproduce the results presented in the paper.

**Comment by nengzechen** [comment]
the .npy given by the author contains 631617 sequences,i wonder if these sequences all come from 11M normal trans or other datas. could you please give me a reply?

**Comment by Bayi-Hu** [comment]
> the .npy given by the author contains 631617 sequences,i wonder if these sequences all come from 11M normal trans or other datas. could you please give me a reply?

Yes. For one account, there might be multiple sequences, thus it may corresponding to multiple embedding vectors. When doing final testing, these multiple vectors will be averaged to generate a final embedding vector.

**Comment by Duongvu05** [comment]
I have tested the code recently, but I cannot reproduce the results presented in the paper for phishing detection w/ fixed training and in/out seperation

**Comment by Bayi-Hu** [comment]
> I have tested the code recently, but I cannot reproduce the results presented in the paper for phishing detection w/ fixed training and in/out seperation

I received feedback from other people that they can reproduce the results, so I’m not sure what the problem is. You can let me know where the problem is.

**Comment by Bayi-Hu** [comment]
> I have tested the code recently, but I cannot reproduce the results presented in the paper for phishing detection w/ fixed training and in/out seperation

Please check Pytorch implementation if the TensorFlow version does not work well. We also have Qiustander's implementation: https://github.com/Qiustander/BERT4ETH_Pytorch. 

and his feedback on reproduction:

<img width="1898" height="134" alt="Image" src="https://github.com/user-attachments/assets/50dacacb-81d4-4540-b189-af7309e21153" />
